build/common.sh

```diff
@@ -142,6 +142,16 @@ function kube::build::verify_prereqs() {
       kube::log::status "Docker Version:"
       "${DOCKER[@]}" version | kube::log::info_from_stdin
     fi
+    if ! kube::build::does_current_builder_support_multiarch; then
+      local sudo=""
+      if [[ $(id -u) != 0 ]]; then
+          sudo=sudo
+      fi
+      ${sudo} "${KUBE_ROOT}/third_party/multiarch/qemu-user-static/register/register.sh" --reset
+      docker buildx rm kube-multiarch-builder || true
+      docker buildx create --use --name=kube-multiarch-builder
```

Comment:
let's leave this for another PR, but noting now:

I think we should probably consider not depending on `--use` in the future. We should also consider trying to avoid the qemu call when it's already setup and/or re-using the kube-multiarch-builder if it already exists.

This is a reasonable start though, and I want to avoid blocking the work depending on this PR from optimizing it further. We mostly care in the case that builds are running concurrently on a host (stop clobbering binfmt_misc unnecessarily since it's global).

Reply:
> I think we should probably consider not depending on --use in the future.
Is this preferred: 
```
docker buildx create --name=kube-multiarch-builder
docker buildx use kube-multiarch-builder
```

Reply:
Er sorry, I mean depend on it being the current global default, not the flag vs the command. We can pass `--builder` to all of the `docker buildx build` calls _if_ we can settle on a name consistently 🙃 

That's a larger refactor we should consider in the future, we already depend on this consistently today, and when we do it we want to consider the other points re: not recreating it unnecessarily (and thus perhaps breaking a call assuming it is available).

Reply:
https://github.com/kubernetes/kubernetes/pull/99080/files#r579393246 is the first more interesting part of that